### PR TITLE
hooks: applypatch-msg

### PR DESCRIPTION
**Pull request by tcely**
When using `git am` add ~~`Merged-By` and~~ `Closes GH-XXX` lines.

Be sure to set `GITHUB_PR_NUMBER` environment variable first.

### CODE REVIEW

File: .githooks/applypatch-msg [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# ensure a blank line when the file is not empty
+test -s "${commit_msg_file}" && _printf '\n\n' >> "${commit_msg_file}"
+
+if test -n "${GITHUB_PR_NUMBER}" ; then
+    _printv GITHUB_PR_KEYWORD "Closes GH-${GITHUB_PR_NUMBER}"
+    _printl "${GITHUB_PR_KEYWORD}" >> "${commit_msg_file}"
+fi
+
+_printv MERGED_BY 'Merged-By:' "$(get_committer_id)"
+_printl "${MERGED_BY}" >> "${commit_msg_file}"
```
[the remark this answers is not in the document]
Comment (tcely) [reply .githooks/applypatch-msg]: I don't know of any special meaning for `Merged-By:` it was just a descriptive label I chose.

I believe this is for interfaces that don't show the committer identity. You should ask @clandmeter since this was requested.

File: .githooks/applypatch-msg [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+# Hook: applypatch-msg
+# This hook is invoked by git-am[1]. It takes a single parameter, the
```
[the remark this answers is not in the document]
Comment (tcely) [reply .githooks/applypatch-msg]: Sure, but having it here will save trouble for anyone who wants to understand how this handler can be extended.

File: .githooks/applypatch-msg (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ikke) [reply .githooks/applypatch-msg]: I just confirmed with clandmeter that the information already in the commit (comitted ident) is enough.

**Comment by Ikke** [comment]
I do like the use of a hook to automatically tag commits (though I noticed that every tagged commit will add another entry in the PR, so not sure if we want to tag all commits).

However I do have the feeling that this script is a little bit over-engineered for it's purpose. It's written as if it needs to be able to run almost anywhere, while it's most likely only run on Alpine. In the mean time, it adds a lot complexity by doing it this way and making it hard to follow.

Do you feel it's necessary doing it this way?

File: .githooks/applypatch-msg [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+#### END sample file ####
+
+git_exec_path="$(git --exec-path)"
+
+if test -r "${git_exec_path}/git-sh-setup" ; then
+    . "${git_exec_path}/git-sh-setup" || exit 1
+fi
+
+require_work_tree_exists &&
```
Comment (Ikke) [inline .githooks/applypatch-msg]: Doesn't the fact that this hook is run by git am mean that a working tree most be present?

Comment (tcely) [reply .githooks/applypatch-msg]: You can try `git am` in a bare repository, even if you should not.

**Comment by tcely** [comment]
Since this runs on the git client side, it is going to need to run in as many places where aports is cloned as possible.

**Comment by Ikke** [comment]
@tcely How likely would someone use this hook somewhere where they don't run abuild. Or in other words, would this hook need to be more portable than abuild?

The hook that I'm currently using is:

```sh
#!/bin/sh

commitfile=$1

if [ -n "$GITHUB_PR" ]; then
    printf '\nCloses GH-%d' "$GITHUB_PR" >>"$commitfile"
fi
```

Of course this might be too simplistic for general use, but it kind of illustrates the amount of 'supporting' code compared to functionality.

**Comment by tcely** [comment]
@Ikke 
> How likely would someone use this hook somewhere where they don't run abuild. Or in other words, would this hook need to be more portable than abuild?

Extremely likely. This hook runs with `git am` and really has no relation to `abuild`. Think about those using `docker-abuild` for instance.

Also, your example hook terminates the chain by not running the `commit-msg` hook.
